developer_guides/virtualisation/virtualisation.rst

```diff
@@ -0,0 +1,255 @@
+which is then forwarded to the DSP firmware. The guest uses the .send_msg()
+callback to forward any IPC, including topology related, with no processing at
+all, to the host. The only exception is blocking IPCs when the guest is
+resuming, while the host doesn't need to be resumed (see 3.3 section on "reset
+count" for details).
+
+The host receives the guest topology over the IPC virtual queue. In most cases
+the host IPC handler just forwards IPCs to the DSP, but in some cases it has to
+handle or modify IPC contents locally. One such example is linking the guest and
+host topology fragments together. As mentioned above, guest topology files
+contain a new token, referencing mixer or demux components in the host topology.
+This value is then assigned to the .ref_comp_id field of struct
+sof_ipc_comp_config during topology file parsing and sent by the guest to the
+host. The host then uses that value to modify connection IPC messages from the
+guest before sending them to the DSP.
+
+3.2. Streaming
+--------------
+
+Guest audio streaming is mostly transparent for the host. The host audio
+subsystem doesn't get involved with most guest streaming or kcontrol operations.
+All those operations are completely processed on the guest and only resulting
+DSP IPCs are forwarded to the host. The only exceptions are beginning and ending
+of the streaming, at which times the ASoC subsystem on the host has to be
+informed, that the respective virtual pipeline, typically including a physical
+codec DAI, is becoming active or inactive respectively. Deactivation of the
+guest pipeline on the host requires changes to the ASoC core, without which the
+guest audio path is not recognised as inactive and is not pruned, thus making
+further streaming by the guest impossible.
+
+Upon reception of the STREAM_TRIG_START IPC message the vhost driver activates
+the virtual PCM pipeline and updates the routing information on the host.
+
+As mentioned above, it is our goal to implement data zero-copy. As long as
+this isn't the case, the guest VirtIO driver implements .block_write() and
+.block_read() SOF DSP operation methods for data streaming via the data
+VirtQueue.
+
+The host receives requests on the data VirtQueue and copies data between PCM
+runtime buffers and the queue and responds either with a status or with a data
+buffer.
+
+Similarly to the data, streaming buffer position updates are currently also
+transferred via a dedicated VirtQueue. Both the host and the guest are
+configured to use IPC messages for position updates. Buffers for VirtIO
+VirtQueues are always provided by guests, therefore there should always be a
+position update buffer waiting on the host side to send a position update
+message as soon as one arrives from the DSP.
+
+Upon reception of the STREAM_TRIG_STOP IPC message the vhost driver updates
+the routing information and deactivates the virtual PCM pipeline on the host.
```

Comment:
again this is far from clear, please clarify why routing information and virtual pipeline refer to.